pkgs/os-specific/linux/applespi/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchgit, pkgs }:
+let
+  version = "git";
+  kernel = pkgs.linux;
+in
+stdenv.mkDerivation {
+  name = "applespi-${version}-${kernel.version}";
+
+  src = fetchgit {
```

Comment:
Prefer `fetchFromGitHub` for GitHub, as it will fetch only the revision by using the GitHub specific ways to fetch a single commit's tree.